alignak/daemons/arbiterdaemon.py

```diff
@@ -764,19 +604,20 @@ def setup_new_conf(self):
 
         :return: None
         """
-        conf = self.new_conf
-        if not conf:
-            return
-        conf = cPickle.loads(conf)
-        self.new_conf = None
-        self.cur_conf = conf
-        self.conf = conf
-        for arb in self.conf.arbiters:
-            if (arb.address, arb.port) == (self.host, self.port):
-                self.me = arb
-                arb.is_me = lambda x: True  # we now definitively know who we are, just keep it.
-            else:
-                arb.is_me = lambda x: False  # and we know who we are not, just keep it.
+        with self.conf_lock:
+            conf = self.new_conf
+            if not conf:
```

Comment:
I didn't had a look at the algorithm. I only added a lock to prevent other thread for resetting conf while setting up. Do yeah,this part could be rewritten maybe in another PR :)